flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/batch/sql/SetOperatorsITCase.scala

```diff
@@ -260,4 +260,108 @@ class SetOperatorsITCase(
     val results = result.toDataSet[Row].collect()
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
+
+  @Test
+  def testInWithFilter(): Unit = {
```

Comment:
Should we validate the feature with plan tests (`TableTestBase`) rather than runtime tests? 
The feature does not require new operators, just expressions which are tested by dedicated tests and joins which are also tested.